test/parallel/test-http-client-input-function.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const common = require('../common');
+const http = require('http');
+const assert = require('assert');
+
+{
+  const server = http.createServer(common.mustCall((req, res) => {
+    res.writeHead(200);
+    res.end('hello world');
+  })).listen(8080, '127.0.0.1');
```

Comment:
Please use port 0 to make this dynamic and use server.address() to get the actual port.

Reply:
thanks, has changed to port 0, and move `ClientRequest` to server `listening` callback